django_project/certification/views/attendee.py

```diff
@@ -73,6 +82,24 @@ def get_form_kwargs(self):
         })
         return kwargs
 
+    def form_valid(self, form):
+        add_to_course = self.request.POST.get('add_to_course')
+        if add_to_course is None:
+            if form.is_valid():
+                form.save()
+        else:
+            if form.is_valid():
+                object = form.save()
+                course_slug = self.kwargs.get('slug', None)
+                course = Course.objects.get(slug=course_slug)
+                course_attendee = CourseAttendee(
+                    attendee=object,
+                    course=course,
+                    author=self.request.user
+                )
+                course_attendee.save()
+        return HttpResponseRedirect(self.get_success_url())
```

Comment:
Just curious, why not letting the django framework do the redirection by itself after the `form_valid` method?

Reply:
Also no need to call the parent `form_valid`? I'm not sure, just asking

Reply:
Because the redirection is based upon the `add_to_course` input. If `add_to_course` is checked, after `form_valid` it will create new course attendee using the newly created attendee object and then it will redirect to `course detail view`. But when `add_to_course` is not checked, after saving new attendee it will redirect to `course attendee create view`.

Reply:
@Gustry , I think by calling form_valid, we make sure that it redirect after form validation and after saving the new objects. cmiiw.

Reply:
Well, if I read your code in this `form_valid`, it will, in any case, hit this return statement L101. Django framework will call by itself the `get_success_url` anyway if you call `return super(YourClassName, self).form_valid(form)` and Django will perform additional task included in the parent method.

Reply:
I see... I have fixed it using `super(AttendeeCreateView, self).form_valid(form)` now. Thank you @Gustry :)